lua/pulsar_lib/core/sql/sv_sql.lua

```diff
@@ -75,8 +75,11 @@ function SQL:prepareStatement(query, values)
 			newQuery = newQuery .. self:Escape(value)
 		elseif type(value) == "boolean" then
 			newQuery = newQuery .. (value and "1" or "0")
-		else
+		elseif tonumber(value) then
 			newQuery = newQuery .. value
+		else
+			ErrorNoHaltWithStack("Invalid value type for prepared statement, expected nil, string, boolean or number, got " .. type(value))
```

Comment:
You should use PulsarLib.Logging.Fatal here

Reply:
.Fatal won't drop the stack, so I'd make two calls, one with the error, and one with debug.traceback() or whatever that function was.